src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1159,6 +1159,11 @@ object Types {
     private[this] var lastSymbol: Symbol = _
     private[this] var checkedPeriod = Nowhere
 
+    /** Provided for debugging, can be used to intercept assignments of uncommon denotations
+     */  
+    private def checkDenot(d: Denotation)(implicit ctx: Context) = {
```

Comment:
I guess this was used for debugging. Do we want to retain it?
If yes, why not in form of setter for `lastDenotation`?